grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1234,10 +1234,8 @@ class StockMovementService {
                         requisitionItem?.product?.productCode,
                 ].toArray(), errorMessage)
 
-                // FIXME: Refactor this part to get rid of need to throw wrapped exception plus consider not allowing zeroing out picked stock
-                // Throw checked exception with ValidationException cause to not rollback deleted picklist while throwing unchecked exception
-                // See: https://pihemr.atlassian.net/browse/OBPIH-5318
-                throw new Exception(errorMessage, new ValidationException(errorMessage, requisitionItem.errors))
+                // FIXME: consider not allowing zeroing out picked stock
+                throw new ValidationException(errorMessage, requisitionItem.errors)
```

Comment:
I'm all for this change. I just don't understand the rollback issue well enough yet.

Reply:
Leaving this for @awalkowiak to review, but I think this might require more scope or a spike to investigate and resolve properly.

Reply:
Ok, so here is the deal. Looks like this is working, but **might be** broken right now. If the user will have a picklist item that won't be available anymore, then if he tries to reload the picklist he will be getting the error about it, but the picklist won't be deleted. I am ok with merging this one, but this: https://pihemr.atlassian.net/browse/OBPIH-5318 needs to be thoroughly retested (plus similar scenarios regarding the validation). 
If the rollback on the deleted picklist while throwing this exception will cause troubles on validation, then we might have to find another way of returning these messages to the API client.

Reply:
Ok. Just take a look at my comments above before merging.